proto/workload.proto

```diff
@@ -74,8 +74,22 @@ message Service {
   repeated string subject_alt_names = 6;
 }
 
+// Workload represents a workload - an endpoint (or collection behind a hostname).
+// The xds primary key is "uid" as defined on the workload below.
+// Secondary (alias) keys are the unique `network/IP` pairs that the workload can be reached at.
 message Workload {
   // UID represents a globally unique opaque identifier for this workload.
+  // For k8s resources, it is recommended to use the more readable format:
+  //
+  // cluster/group/version/kind/namespace/name/optional-index
```

Comment:
```suggestion
  // cluster/group/kind/namespace/name/section-name
```

WorkloadEntry v1alpha3 and WE v1beta1 are the same resource.

I would use "section name" to align with Gateway API naming (not that its important, as this is just internal doc)